sdks/java/fn-execution/src/main/java/org/apache/beam/sdk/fn/stream/DirectStreamObserver.java

```diff
@@ -59,9 +60,8 @@ public DirectStreamObserver(Phaser phaser, CallStreamObserver<T> outboundObserve
 
   @Override
   public void onNext(T value) {
```

Comment:
So this function could be called in a multiple threading environment thus we need an atomic integer?

Reply:
Yes, I saw Luke mention on some other thread these observers are supposed to be threadsafe. An example is the BeamFnControlClient which calls this from different threads on the executor.  And this class is marked @Threadsafe.